pkg/azurePipelines/azurePipelines.go

```diff
@@ -83,6 +84,36 @@ func (p *AzurePipelines) loadConfig(deployType string) (*config.DraftConfig, err
 	return &draftConfig, nil
 }
 
+func (p *AzurePipelines) overrideFilename(draftConfig *config.DraftConfig, srcDir string) error {
+	if draftConfig.FileNameOverrideMap == nil {
+		draftConfig.FileNameOverrideMap = make(map[string]string)
+	}
+	pipelineVar, err := draftConfig.GetVariable(pipelineNameVar)
+	if err != nil {
+		return fmt.Errorf("error getting pipeline name variable: %w", err)
+	}
+
+	files, err := fs.ReadDir(p.pipelineTemplates, srcDir)
+	if err != nil {
+		return fmt.Errorf("error reading source directory: %w", err)
+	}
+
+	for _, f := range files {
+		if f.Name() == configFileName {
+			continue
+		}
+		if f.IsDir() {
+			if err := p.overrideFilename(draftConfig, path.Join(srcDir, f.Name())); err != nil {
+				return fmt.Errorf("error overriding filename: %w", err)
+			}
+			continue
+		}
+
+		draftConfig.FileNameOverrideMap[f.Name()] = pipelineVar.Value + ".yaml"
```

Comment:
this names every file to `pipelineVar.Value + ".yaml"` right? if there were multiple files then wouldn't they collide?

Reply:
That's right, I kept this logic since we have one pipeline file per deploy type. In case more files are added, I've changed the logic to check for the pipeline template filename while overriding.